mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -251,8 +265,198 @@ static void populateDefaults() {
                    paddingWidth.getValue(), strideWidth.getValue()));
 }
 
-static LogicalResult populateHostHarnessLogic(ModuleOp &module, OpBuilder &builder,
-                                              MLIRContext &context, mlir::FloatType dataType) {
+static void populateHostValidationLogic(
+    ModuleOp &module, OpBuilder &builder, MLIRContext &context,
+    mlir::FloatType dataType, Block *block, mlir::Value &ref2MemsetValue,
+    mlir::FuncOp &mcpuMemset4DFuncOp, SmallVector<int64_t, 4> &filterDimension,
+    SmallVector<int64_t, 4> &inputDimension,
+    SmallVector<int64_t, 4> &outputDimension, mlir::AllocOp &filterHostAllocOp,
+    mlir::AllocOp &inputHostAllocOp, mlir::AllocOp &outputHostAllocOp) {
+  // Emit CPU alloc
```

Comment:
Could you populate the host validation as a separate function?

Reply:
Then the host validation function and the host harness function would have lots of duplicated code

Reply:
It doesn't have to be duplicated. If the validation routine overall is a separate one, then the only thing need to add to the host harness is a invocation to the validation.

I'd suggest you to change the `conv2d_harness_cpu_verifier.mlir` as a start. Once we reach at the agreement there on the prototype, we can use it to guide the discussion here.

Reply:
Sure. I can start with conv2d_harness_cpu_verifier.mlir.